src/Markdown.MAML/Renderer/Markdownv2Renderer.cs

```diff
@@ -70,6 +78,100 @@ private string MamlModelToString(MamlCommand mamlCommand, Hashtable yamlHeader,
+            }
+
+            //processes all nodes in order
+            foreach (var currentNode in document.Children)
+            {
+                switch (currentNode.NodeType)
+                {
+                    case MarkdownNodeType.Paragraph:
+                        ParagraphNode paragraphNode = currentNode as ParagraphNode;
+                        AddAboutParagraph(paragraphNode);
+                        break;
+                    case MarkdownNodeType.Heading:
+                        HeadingNode headingNode = currentNode as HeadingNode;
+                        AddAboutHeading(headingNode, document);
+                        break;
+                    case MarkdownNodeType.CodeBlock:
+                        CodeBlockNode codeblockNode = currentNode as CodeBlockNode;
+                        AddAboutCodeBlock(codeblockNode);
+                        break;
+                }
+            }
+            
+            return RenderCleaner.FullNormalization(_stringBuilder.ToString());
+        }
+
+        private void AddAboutCodeBlock(CodeBlockNode codeblockNode)
+        {
+            string[] lines = codeblockNode.Text.Split(new string[] {"\r\n"}, StringSplitOptions.None);
+            
+            foreach (string line in lines)
+            {
+                _stringBuilder.Append("    ");
+                _stringBuilder.AppendLine(line);
+            }
+
+            _stringBuilder.AppendLine();
+        }
+
+        private void AddAboutHeading(HeadingNode headingNode, DocumentNode document)
+        {
+            if (document.Children.ElementAt(0) as HeadingNode == headingNode)
+            {
+                _stringBuilder.AppendLine("TOPIC");
+            }
+            else if (headingNode.HeadingLevel == 1)
+            {
+                _stringBuilder.AppendLine(headingNode.Text.ToUpper());
+            }
+            else if (headingNode.HeadingLevel == 2 && document.Children.ElementAt(1) == headingNode)
+            {
+                _stringBuilder.AppendLine("    " + headingNode.Text.ToLower() + "\r\n");
+            }
+            else if (headingNode.HeadingLevel == 2)
+            {
+                _stringBuilder.AppendLine(headingNode.Text);
+            }
+            else
+            {
+                _stringBuilder.AppendLine("    " + headingNode.Text.ToUpper());
+            }
+        }
+
+        private void AddAboutParagraph(ParagraphNode paragraphNode)
+        {
+            foreach (string lineContent in paragraphNode.Spans.Select(span => span.Text))
+            {
+                //handles all paragraph lines over 80 characters long and not headers
+                if (lineContent.Length > _maxLineWidth - 4)
+                {
+                    WrapAndAppendLines(lineContent, _stringBuilder);
+                    _stringBuilder.AppendLine();
+                }
+                else if (lineContent == "\n")
```

Comment:
don't use == for string comparison. Use StringComparer.OrdinalIgnoreCase.Equals()